jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -654,32 +697,299 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
+            If requested (see ``return_bad_regions`` option), return a
+            list of ``(region, reason)`` tuples for region objects that failed to load.
+            If all the regions loaded successfully, this list will be empty.
+            If not requested, return `None`.
+
+        """
+        if len(self.app.data_collection) == 0:
+            raise ValueError('Cannot load regions without data.')
+
+        from photutils.aperture import (CircularAperture, SkyCircularAperture,
+                                        EllipticalAperture, SkyEllipticalAperture,
+                                        RectangularAperture, SkyRectangularAperture,
+                                        CircularAnnulus, SkyCircularAnnulus)
+        from regions import (Regions, CirclePixelRegion, CircleSkyRegion,
+                             EllipsePixelRegion, EllipseSkyRegion,
+                             RectanglePixelRegion, RectangleSkyRegion,
+                             CircleAnnulusPixelRegion, CircleAnnulusSkyRegion)
+        from jdaviz.core.region_translators import regions2roi, aperture2regions
+
+        if not isinstance(regions, (list, tuple, Regions, SpectralRegion)):
+            regions = [regions]
+
+        n_loaded = 0
+        bad_regions = []
+
+        # To keep track of masked subsets.
+        msg_prefix = 'MaskedSubset'
+        msg_count = _next_subset_num(msg_prefix, self.app.data_collection.subset_groups)
+
+        viewer_parameter = kwargs.pop('viewer', None)
+        viewer_name = viewer_parameter or list(self.app._jdaviz_helper.viewers.keys())[0]
+        viewer = self.app.get_viewer(viewer_name)
+        # Subset is global but reference data is viewer-dependent.
+        if refdata_label is None:
+            data = viewer.state.reference_data
+        else:
+            data = self.app.data_collection[refdata_label]
+
+        # TODO: Make this work for data cube.
+        # https://github.com/glue-viz/glue-astronomy/issues/75
+        has_wcs = data_has_valid_wcs(data, ndim=2)
+
+        combo_mode_is_list = isinstance(combination_mode, list)
+        if combo_mode_is_list and len(combination_mode) != (len(regions)):
+            raise ValueError("list of mode must be size of regions")
+
+        for index, region in enumerate(regions):
+            # Set combination mode for how region will be applied to current subset
+            # or created as a new subset
+            if combo_mode_is_list:
+                combo_mode = combination_mode[index]
+            else:
+                combo_mode = combination_mode
+
+            if combo_mode == 'new':
+                # Remove selection of subset so that new one will be created
+                self.app.session.edit_subset_mode.edit_subset = None  # No overwrite next iteration
+                self.app.session.edit_subset_mode.mode = SUBSET_MODES_PRETTY['new']
+            elif combo_mode:
+                self.combination_mode.selected = combo_mode
```

Comment:
so if a list of regions is passed with a list of combination modes but an entry is invalid, the previous entries will be loaded and then an error will be raised here that it is not valid?  Can/should we validate the entire list in advance (possibly as follow-up if necessary)?

Reply:
Validate the list of combination modes? Yes that should be a quick addition.